### PR TITLE
Feat/move resources

### PR DESCRIPTION
**Pull request by gweiying**
This PR adds an endpoint for moving resources from one resource room to another. Similar to the move endpoints for collection pages and pages, it takes in an array of files and iteratively deletes and creates new files. 

This PR also adds simple refactoring to the resourcePages routes to handle Base64 encoding and decoding on the backend, extending #138.

### CODE REVIEW

**Comment by kwajiehao** [comment]
As discussed offline, since there are no use cases for bulk movement of resource pages between resource categories, we will be modifying this feature to move only single resource pages from one category to another

**Review by kwajiehao — CHANGES_REQUESTED** [review]
Typo in route params

File: middleware/auth.js [hunk 1]
```diff
@@ -125,6 +126,7 @@ auth.get('/v1/sites/:siteName/resources', verifyJwt)
 auth.post('/v1/sites/:siteName/resources', verifyJwt)
 auth.delete('/v1/sites/:siteName/resources/:resourceName', verifyJwt)
 auth.post('/v1/sites/:siteName/resources/:resourceName/rename/:newResourceName', verifyJwt)
+auth.post('/v1/sites/:siteName/resources/:resourceRoomName/move/:newResourceRoomName', verifyJwt)
```
[the remark this answers is not in the document]
Comment (gweiying) [reply middleware/auth.js]: good catch, fixed here

**Review by kwajiehao — APPROVED** [review]
LGTM!

**Pull request merged**